libs/partners/pinecone/pyproject.toml

```diff
@@ -14,12 +14,14 @@ license = "MIT"
 # <3.13 is due to restriction in pinecone-client package
 python = ">=3.8.1,<3.13"
 langchain-core = ">=0.1.52,<0.3"
-pinecone-client = "^3.2.2"
+pinecone-client = "^4.1.1"
 # Support Python 3.8 and 3.12+.
 numpy = [
     {version = "^1", python = "<3.12"},
     {version = "^1.26.0", python = ">=3.12"}
 ]
+pinecone-plugin-inference = ">=0.3.0,<1.0.0"
```

Comment:
is this used anywhere? While in preview, can we make this an optional dependency, and I assume once it's out of preview it'll be in `pinecone-client`?

Reply:
hi @efriis , yes this is used for the `PineconeEmbeddings` class, it integrates directly into the `pinecone-client` library so you won't see it being imported anywhere, but if it is not installed the `PineconeEmbeddings` won't work

Reply:
Up to you. Depending on stability, I would probably recommend making it an optional dependency, with a nice ImportError instructing users to `pip install pinecone-plugin-inference` if it doesn't work.

Here's the rough setup of importing the optional with that https://github.com/langchain-ai/langchain/blob/master/libs/community/langchain_community/vectorstores/pinecone.py#L29-L35